adr/ADR-47.md

```diff
@@ -0,0 +1,122 @@
+# Request Many
+
+| Metadata | Value                      |
+|----------|----------------------------|
+| Date     | 2024-09-26                 |
+| Author   | @aricart, @scottf, @Jarema |
+| Status   | Partially Implemented      |
+| Tags     | client                     |
+
+| Revision | Date       | Author    | Info                    |
+|----------|------------|-----------|-------------------------|
+| 1        | 2024-09-26 | @scottf   | Document Initial Design |
+
+## Problem Statement
+Have the client support receiving multiple replies from a single request, instead of limiting the client to the first reply,
+and support patterns like scatter-gather and sentinel.
+
+## Basic Design
+
+The user can provide some configuration controlling how and how long to wait for messages.
+The client handles the requests and subscriptions and provides the messages to the user.
+
+* The client doesn't assume success or failure - only that it might receive messages.
+* The various configuration options are there to manage and short circuit the length of the wait, 
+and provide the user the ability to directly stop the processing.
+* Request Many is not a recoverable operation, but it could be wrapped in a retry pattern.
+* The client should communicate status whenever possible, for instance if it gets a 503 No Responders
+
+## Config
+
+### Total timeout
+
+The maximum amount of time to wait for responses. When the time is expired, the process is complete.
+The wait for the first message is always made with the total timeout since at least one message must come in within the total time.
+
+* Always used
+* Defaults to the connection or system request timeout.
+
+### Stall timer
+
+The amount time to wait for messages other than the first (subsequent waits). 
+Considered "stalled" if this timeout is reached, indicating the request is complete.
+
+* Optional
+* Less than 1 or greater than or equal to the total timeout behaves the same as if not supplied.
+* Defaults to not supplied.
+* When supplied, subsequent waits are the lesser of the stall time or the calculated remaining time. 
+This allows the total timeout to be honored and for the stall to not extend the loop past the total timeout.
+
+### Max messages
+
+The maximum number of messages to wait for. 
```

Comment:
Depending on the implementation approach, with new style and max responses traffic wise you can get more than the max responses anyway and the client will receive them and not process them.  With old style and auto unsub you can short circuit the number of max messages to avoid flooding the client with unnecessary responses, some pros and cons depending on the protocols sent.  
New style/mux based approach has more value added though since more difficult to maintain for users on their own.